cachelib/allocator/CacheAllocator-inl.h

```diff
@@ -1697,12 +1533,25 @@ CacheAllocator<CacheTrait>::findEviction(TierId tid, PoolId pid, ClassId cid) {
             chainedItem_
                 ? &toRecycle_->asChainedItem().getParentItem(compressor_)
                 : nullptr;
+        auto l_ = chainedItem_ 
```

Comment:
I don;t think we should take a lock without checking if the item is chained first - it will have significant overhead I think